cocotb/drivers/avalon.py

```diff
@@ -567,19 +681,25 @@ def _send_string(self, string, sync=True):
         word = BinaryValue(bits=len(self.bus.data),
                            bigEndian=self.config['firstSymbolInHighOrderBits'])
 
-        empty  = BinaryValue(bits=len(self.bus.empty), bigEndian=False)
         single = BinaryValue(bits=1, bigEndian=False)
-
+        if self.use_empty:
+            empty = BinaryValue(bits=len(self.bus.empty), bigEndian=False)
 
         # Drive some defaults since we don't know what state we're in
-        # self.bus.empty <= 0
-        yield NextTimeStep()
+        if self.use_empty:
+            self.bus.empty <= 0
         self.bus.startofpacket <= 0
         self.bus.endofpacket <= 0
         self.bus.valid <= 0
         if hasattr(self.bus, 'error'):
             self.bus.error <= 0
 
+        if hasattr(self.bus, 'channel'):
+            self.bus.channel <= 0
+            channel_value = BinaryValue(bits=len(self.bus.channel), bigEndian=False)
```

Comment:
Since you don't use it anywhere else, it would probably make sense to move this line down to line 769

Reply:
And bits was just deprecated and is now called 'n_bits'.